instance/instance_external_test.go

```diff
@@ -43,11 +43,16 @@ func TestGetInstancesReturnsOK(t *testing.T) {
 			},
 		}
 
-		instanceAPI := &instance.Store{Host: "http://lochost://8080", Storer: mockedDataStore}
+		auditor := audit_mock.New()
+		instanceAPI := &instance.Store{Host: "http://lochost://8080", Storer: mockedDataStore, Auditor: auditor}
 		instanceAPI.GetList(w, r)
 
 		So(w.Code, ShouldEqual, http.StatusOK)
 		So(len(mockedDataStore.GetInstancesCalls()), ShouldEqual, 1)
+		auditor.AssertRecordCalls(
+			audit_mock.Expected{instance.GetInstancesAction, audit.Attempted, nil},
```

Comment:
So `audit_mock.Expected` method, `composite literal uses unkeyed fields`.

It is safer to explicitly pair up keys to values instead of relying on order of struct, by doing the following instead:
`audit_mock.Expected{Action: instance.GetInstancesAction, Result: audit.Attempted, Params: nil}`

If the struct order is changed; for instance we want to alphabetically order the struct from:
```
type Expected struct {
	Action string
	Result string
	Params common.Params
}
```
to:
```
type Expected struct {
	Action string
	Params common.Params
	Result string
}
```
Then the tests will break as we have unkeyed fields in our test

Reply:
Fair point. Personally I see this as a flaw in Golang's approach to struct initialisation. Having a constructor with named / ordered parameters solves this problem... I know you can write a constructor function if you want but it's not enforced. Like most things in Go it make its your problem :-p